### ISSUE TITLE
Develop a strategy regarding backwards compatibility

### ISSUE THREAD

**Issue by Niederb** [scs/substrate-api-client#788]
We should discuss and document how/if we want to support breaking changes (such as #777) by providing backwards compatibility. Some considerations:
- The code gets cluttered quickly if there are lots of conditional compilation instructions
- Running the testing for each option gets out of hand quickly
  - If one is strict we would also need to run our tests for each supported polkadot version
- Different chains (polkadot, kusama, rococo) might have a different configuration and therefore it would be important for our client

**Comment by Niederb** [comment]
I don't think we need to put much effort into this, but it would be nice to document somewhere what we do. It is also not clear to me when we should create a release. Maybe something like:
- Always create a release for a new polkadot release
  - No new releases for polkadot bugfix releases unless there are known issues
- Do not backport features and bugfixes per default. If requested we decide on a case by case basis

**Comment by haerdib** [comment]
Regarding the current backport with the Merkelized Metadata : This would mean that we remove the build flag in the next polkadot release. Correct?

**Comment by Niederb** [comment]
> Regarding the current backport with the Merkelized Metadata : This would mean that we remove the build flag in the next polkadot release. Correct?

If I understand correctly it is still possible to run a node without this feature:
https://github.com/polkadot-fellows/runtimes/blob/afc36ca73146d7e22887fb11c6631a7129d68dd2/relay/kusama/build.rs#L17
Is the check already deployed and active on polkadot?

> Totally so. Would the current release strategy with marked breaking changes suffice? ( see releases in https://github.com/scs/substrate-api-client/releases)

Yes, for a specific release this should suffice. I would still mention it in our backwards compatibility "strategy" though.

**Comment by Niederb** [comment]
See #809
